SimPEG/EM/FDEM/RxFDEM.py

```diff
@@ -1,37 +1,59 @@
-import SimPEG
+import properties
 
+from ...Survey import BaseRx
 
+__all__ = [
+    'Point_e', 'Point_b', 'Point_h', 'Point_j', 'Point_bSecondary'
+]
 
-class BaseRx(SimPEG.Survey.BaseRx):
-    """
-    Frequency domain receiver base class
 
-    :param numpy.ndarray locs: receiver locations (ie. :code:`np.r_[x,y,z]`)
-    :param string orientation: receiver orientation 'x', 'y' or 'z'
-    :param string component: real or imaginary component 'real' or 'imag'
+class BaseFDEMRx(BaseRx):
+    """
+    Base frequency domain electromagnetic receiver. Inherit this to build a
+    frequency domain electromagnetic receiver.
     """
 
-    def __init__(self, locs, orientation=None, component=None):
-        assert(
-            orientation in ['x', 'y', 'z']
-        ), "Orientation {0!s} not known. Orientation must be in 'x', 'y', 'z'."
-        " Arbitrary orientations have not yet been implemented.".format(
-            orientation
-        )
-        assert(
-            component in ['real', 'imag']
-            ), "'component' must be 'real' or 'imag', not {0!s}".format(
-                component
-            )
-
-        self.projComp = orientation
-        self.component = component
-
-        # TODO: remove rxType from baseRx
-        SimPEG.Survey.BaseRx.__init__(self, locs, rxType=None)
+    # TODO: eventually, this should be a Vector3 and we can allow arbitraty
+    # orientations
+    orientation = properties.StringChoice(
+        "orientation of the receiver 'x', 'y', or 'z'",
+        choices=['x', 'y', 'z'],
+        required=True
+    )
+
+    component = properties.StringChoice(
```

Comment:
@sgkang, @dougoldenburg, @rowanc1, @grosenkj: I would be curious to hear your thoughts on the name for this key-word argument. `component` is perhaps easily confused with spatial component (`orientation`), but I cannot think of another more explicit name. 
```
rx_bz_real = FDEM.Rx.Point_b(orientation='z', component='real') 
rx_bz_imag = FDEM.Rx.Point_b(orientation='z', component='imag') 
```